workers/datasets_based/src/datasets_based/workers/first_rows.py

```diff
@@ -526,24 +570,18 @@ def compute_first_rows_response(
             "Server error while post-processing the split rows. Please report the issue.",
             cause=err,
         ) from err
-    # get the size of the surrounding JSON (without the rows)
-    features_list = to_features_list(features=features)
-    response: FirstRowsResponse = {
-        "dataset": dataset,
-        "config": config,
-        "split": split,
-        "features": features_list,
-        "rows": [],
-    }
-    surrounding_json_size = get_json_size(response)
+
     # truncate the rows to fit within the restrictions, and prepare them as RowItems
+    surrounding_json_size = get_json_size(response)
```

Comment:
```suggestion
    surrounding_json_size = get_json_size(response_features_only)
```

also: maybe try to avoid computing the size twice.